modules/dnn/src/layers/reshape_layer.cpp

```diff
@@ -182,11 +182,39 @@ class ReshapeLayerImpl : public ReshapeLayer
         return true;
     }
 
+    bool forward_ocl(InputArrayOfArrays inps, OutputArrayOfArrays outs, OutputArrayOfArrays internals)
+    {
+        std::vector<UMat> inputs;
+        std::vector<UMat> outputs;
+
+        inps.getUMatVector(inputs);
+        outs.getUMatVector(outputs);
+
+        for (size_t i = 0; i < inputs.size(); i++)
+        {
+            UMat srcBlob = inputs[i];
+            void *src_handle = inputs[i].handle(ACCESS_READ);
+            void *dst_handle = outputs[i].handle(ACCESS_WRITE);
+            if (src_handle != dst_handle)
+            {
+                MatShape outShape = shape(outputs[i]);
+                UMat umat = srcBlob.reshape(1, (int)outShape.size(), &outShape[0]);
+                umat.copyTo(outputs[i]);
```

Comment:
`.assign()`?

Reply:
Do you mean assign the "umat" to OutputArrayOfArrays  ?  here OutputArrayOfArrays is actually a vector of umat, so I can't assign one umat to a vector.  It would be convenient if OutputArrayOfArrays  provides an interface like "assign(UMat umat, int index)" to assign one umat into a vector with  a index.

Reply:
There are `.assign()` [overloads](https://github.com/opencv/opencv/pull/10126/files) with `std::vector<UMat>`.

Reply:
added  .assign() to overload std::vector<UMat>